profiler/src/ProfilerEngine/Datadog.Profiler.Native/LibddprofExporter.cpp

```diff
@@ -249,15 +249,22 @@ ddog_Endpoint LibddprofExporter::CreateEndpoint(IConfiguration* configuration)
         // Agent mode
 
 #if _WINDOWS
-        bool useDefaultDomainSocket = false;
+        std::string socketPath;
+        const std::string& namePipeName = configuration->GetNamedPipeName();
+        if (!namePipeName.empty())
+        {
+            socketPath = R"(\\.\pipe\)" + namePipeName;
+        }
+        std::string agentUrl = "windows:" + socketPath;
 #else
-        std::error_code ec; // fs::exists might throw if no error_code parameter is provided
-        bool useDefaultDomainSocket = fs::exists("/var/run/datadog/apm.socket", ec);
+        std::string socketPath = "/var/run/datadog/apm.socket";
+        std::string agentUrl = "unix://" + socketPath;
 #endif
 
-        if (useDefaultDomainSocket)
+        std::error_code ec; // fs::exists might throw if no error_code parameter is provided
+        if (fs::exists(socketPath, ec))
```

Comment:
I don't think this is correct for named pipes. If the customer has selected named pipes (by adding a trace name), it doesn't matter if it exists, we should still use it, not fallback to HTTP (as is currently happening I think)